core/src/main/java/io/questdb/cairo/TableReader.java

```diff
@@ -648,8 +648,16 @@ private void copyColumns(
     }
 
     private void copyOrRenewSymbolMapReader(SymbolMapReader reader, int columnIndex) {
-        if (reader != null && reader.isDeleted()) {
-            reader = reloadSymbolMapReader(columnIndex, reader);
+        if (reader != null) {
```

Comment:
is there a way to siplify the code?

`copyOrRenewSymbolMapReader` lacks `copy` action. It reloads symbol conditionally using `reloadSymbolMapReader`.

Both `copyOrRenewSymbolMapReader` and `reloadSymbolMapReader` are called only from `reshuffleSymbolMapReaders` and they overlap on functionality

Reply:
ok, refactored in [6ade81c](https://github.com/questdb/questdb/pull/3787/commits/6ade81cc57825cfd169d6e53340fa52d6ae4e15e)